architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,296 @@
+# Radius Applications RP Component Threat Model
+
+- **Author**: @nithyatsu
+
+## Overview
+
+This document provides a threat model for the Radius Applications RP component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+## Terms and Definitions
+
+| Term                  | Definition                                                                                                                                                                                  |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| RP                 | Resource Provider    |
+| UCP                 | Universal Control Plane for Radius   |
+| DE | Deployment Engine |
+
+## System Description
+
+Applications RP is a Radius control-plane microservice that acts as resource provider for application and its resources. When users deploy an application using Radius, Applications RP manages the lifecycle of the created resources on the user's behalf. 
+
+- Deploying an application may launch the user application's code on the same cluster as Radius, or a different cluster. 
```

Comment:
How do we achieve this? By using workspaces?

Reply:
yes, workspace has kube context as an entry.